### PR TITLE
fix(Skeleton): height/width prop incorrect typing

### LINKED ISSUE #3799: Skeleton height/width prop incorrect typing
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Explain what you did

I'm developing with TypeScript and I'm trying to make the Skeleton component fill up a containing View by setting its height and width to 100%.

### Expected behavior

I expected that setting the `height` and `width` props to the string `"100%"` would be accepted by TypeScript.

### Describe the bug

The `height` and `width` props are typed as numbers so TypeScript complains. When running the code anyway, the height and width is still properly set despite this. As a workaround to the typing issues, you can set the height and width using the `style` prop instead. Interesting, the `style` prop is not listed in [the documentation](https://reactnativeelements.com/docs/components/skeleton#props) but the `skeletonStyle` prop is. However, the `skeletonStyle` prop does not work for this workaround.

### Steps To Reproduce

While using TypeScript, try using the Skeleton component like this: 
```Markdown
<Skeleton width="100%" height="100%" />
```

### Screenshots

_No response_

### Your Environment

<details>
<summary>`npx @rneui/envinfo`</summary>
Sorry, the command is taking a ridiculously long time for some reason. If this is really needed for some reason, I'll edit it in later.
</details>

### PR DESCRIPTION
**Pull request by deepak-rawat78**
Fixes #3799

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation using `yarn docs-build-api`
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/base/src/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -19,12 +19,12 @@ export interface SkeletonProps extends ViewProps {
   /**
    * Width of Skeleton View
    */
-  width?: number;
+  width?: number | string;
```
Comment (khushal87) [inline packages/base/src/Skeleton/Skeleton.tsx]: I would rather assign it to `ViewStyle["width"]` and `ViewStyle["height"]` for future safety.

File: packages/base/src/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -19,12 +19,12 @@ export interface SkeletonProps extends ViewProps {
...
   /**
    * Height of Skeleton View
    * @default 12
    */
-  height?: number;
+  height?: number | string;
```
Comment (khushal87) [inline packages/base/src/Skeleton/Skeleton.tsx]:
```suggestion
  height?: ViewStyle["height"];
```

File: packages/base/src/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (deepak-rawat78) [reply packages/base/src/Skeleton/Skeleton.tsx]: Thanks for suggestion. I have updated these accordingly.

**Review by deepak-rawat78 — COMMENTED** [review]
Types of height and width in Skeleton.tsx updated as per suggestion.

**Comment by deepak-rawat78** [comment]
@khushal87 As per your suggestion, I have updated types of height and width in Skeleton.tsx.

File: packages/base/src/Skeleton/Skeleton.tsx [hunk 3]
```diff
@@ -38,6 +38,10 @@ export interface SkeletonProps extends ViewProps {
    * @type React Component
    */
   LinearGradientComponent?: React.ComponentType<any>;
+  /**
+   * Custom style for skeleton container
+   */
+  style: StyleProp<ViewStyle>;
```
Comment (khushal87) [inline packages/base/src/Skeleton/Skeleton.tsx]: `style` is not needed, in my opinion, since `SkeletonProps` already extends `ViewProps`.

Comment (deepak-rawat78) [reply packages/base/src/Skeleton/Skeleton.tsx]: Yes. Props Updated.

**Review by khushal87 — APPROVED** [review]

**Pull request merged**